GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,169 @@
+                                   userInfo:nil];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:error];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.code, error.code);
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+}
+
+- (void)testMaybeFetchTokenWithAuthFlowError {
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                       code:kGIDSignInErrorCodeUnknown
+                                   userInfo:nil];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] initWithAuthState:nil
+                                                           error:error
+                                                      emmSupport:nil
+                                                     profileData:nil];
+  GIDAuthorizationResponseHandler *responseHandler = 
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:kEMMVersion
+                                                                    flowName:GIDFlowNameSignIn
+                                                               configuration:nil
+                                                                       error:nil];
+
+  [responseHandler maybeFetchToken:authFlow];
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertNil(authFlow.emmSupport);
+}
+
+- (void)testMaybeFetchTokenWithExpirationError {
+  OIDAuthState *authState = [OIDAuthState testInstance];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] initWithAuthState:authState
+                                                           error:nil
+                                                      emmSupport:nil
+                                                     profileData:nil];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:kEMMVersion
+                                                                    flowName:GIDFlowNameSignIn
+                                                               configuration:nil
+                                                                       error:nil];
+
+  [responseHandler maybeFetchToken:authFlow];
+  XCTAssertNotNil(authFlow.authState);
```

Comment:
If we do expect an `authState`, then let's assert that it has what we expect on it.

Reply:
Done.